single.c

```diff
@@ -21,7 +20,7 @@ run_single (task_t *task, config_t *config)
   data.initialized = 0;
   st_context_t st_context = {
     .hash = config->hash,
-    .data = &data,
+    .data = data,
```

Comment:
No need to allocate one more instance of struct crypt_data only for initialisation of one field. This struct is 128kb and we need to initialise only one byte. The proper way to do this:
.data = { .initialized = 0 },